### DISCUSSION TITLE
Why snponly.efi is faster than ipxe.efi when download file?

### DISCUSSION THREAD

**Discussion by ccy** [ipxe/ipxe#425] [Q&A]
All the while I use ipxe.efi to boot into PXE.  After I try the `snponly.efi`, I found the download speed via HTTP protocol is faster in general.  What is the reason behind?

**Comment by NiKiZe** [comment]
Is it the same build and same URL?
You should at least include output from ifstat, or information about driver used in the ipxe.efi case.

iPXE has multiple drivers, the native driver might be slower in some cases, or maybe is some kind of interference, it needs to be debugged quite in depth to answer. But at least we must know which hardware (nic) that you are experiencing this on.

**Reply by ccy** [reply]
Yes. Same source for `snponly.efi` and `ipxe.efi`.

This is the `ifstat` for `ipxe.efi`:

```
net0: 12:34:56:78:90:ab using i219v-10 on 0000:00:1f.6 (Ethernet) [open]
  [Link: up, TX:12 TXE:2 RX:3925 RXE:3644]
  [TXE: 1 x "Network unreachable (http://ipxe.org/28086090)"]
  [TXE: 1 x "Operation not supported (http://ipxe.org/3c654083)"]
  [RXE: 4 x "Invalid argument (http://ipxe.org/1c056082)"]
  [RXE: 3456 x "The socket is not connected (http://ipxe.org/380f6093)"]
  [RXE: 33 x "Operation not supported (http://ipxe.org/3c046083)"]
  [RXE: 148 x "Operation not supported (http://ipxe.org/3c046083)"]
```

For `snponly.efi`, the `ifstat` shows `NII on NII-0000:00:1f.6`
